docker/mongodb-kubernetes-tests/tests/search/search_enterprise_tls.py

```diff
@@ -247,3 +285,42 @@ def get_user_sample_movies_helper(mdb):
             get_connection_string(mdb, USER_NAME, USER_PASSWORD), use_ssl=True, ca_path=get_issuer_ca_filepath()
         )
     )
+
+
+def assert_search_service_prometheus_port(mdbs: MongoDBSearch, should_exist: bool, expected_port: int = 9946):
+    service_name = f"{mdbs.name}-search-svc"
+    service = get_service(mdbs.namespace, service_name)
+
+    assert service is not None, f"Service {service_name} should exist"
+
+    ports = {p.name: p.port for p in service.spec.ports}
+
+    if should_exist:
+        assert "prometheus" in ports, "Prometheus port should be in service when prometheus is enabled"
+        assert (
+            ports["prometheus"] == expected_port
+        ), f"Prometheus port should be {expected_port} but was {ports['prometheus']}"
+    else:
+        assert "prometheus" not in ports, "Prometheus port should not be in service when prometheus is disabled"
+
+
+def assert_search_pod_prometheus_endpoint(mdbs: MongoDBSearch, should_be_accessible: bool, port: int = 9946):
+    pod_name = f"{mdbs.name}-search-0"
+    url = f"http://localhost:{port}/metrics"
```

Comment:
If we hit `http://{mdbs.name}-search-svc:{port}` directly we could omit explicitly asserting properties of the Service object in `assert_search_service_prometheus_port()` altogether, I think. I don't know if it's worth it to maintain the distinction between "the port was configured on the Service" and "prometheus was enabled in mongot configuration" since the former is already tested in the search reconcile helper unit test.

Reply:
Agree 💯 I've reworked it a bit. I'm deploying a tools pod (community image like in snippets) to connect to the probe. Might seem like an overkill for now, but I wanted to take that opportunity and test this approach in practice as we're planning to get rid of the e2e pod entirely and we'll need some sort of proxy for the connectivity checks soon anyway.